lib/Server.js

```diff
@@ -130,12 +130,13 @@ class Server {
         next(err);
       });
 
-      return new Promise(resolve => {
+      const listenPromise = new Promise(resolve => {
         this.httpServer = this.server.listen(this.port, () => {
           console.log(`Server listening at http://${this.hostname}:${this.port}`);
           resolve();
         });
       });
+      return Promise.all([listenPromise, readyPromise]);
```

Comment:
Through config. But I bow out here. Good luck.
On Fri, Nov 27, 2015 at 11:47 PM Philipp von Weitershausen <
notifications@github.com> wrote:

> In lib/Server.js
> <https://github.com/mjohnston/react-native-webpack-server/pull/144#discussion_r46076347>
> :
>
> >          this.httpServer = this.server.listen(this.port, () => {
> >            console.log(`Server listening at http://${this.hostname}:${this.port}`);
> >            resolve();
> >          });
> >        });
> > +      return Promise.all([listenPromise, readyPromise]);
>
> My understanding is the Promise will return immediately. And unless the
> async/await pattern, or generator functions are used, the net result will
> be the same WRT timing of the two servers starting.
>
> You seem to be misunderstanding how promises work. There's a difference
> between *returning* a promise (which usually happens immediately) and
> *resolving* the promise (which usually happens whenever you'd invoke a
> callback indicating success of an asynchronous action).
>
> When constrained to ES5 idioms, this is how you'd use a promise:
>
> somethingThatReturnsAPromise().then(function (result) {
>   doSomethingWith(result);
> });
>
> An equivalent, more modern usage of promises would be this:
>
> let result = await somethingThatReturnsAPromise();doSomethingWith(result);
>
> Functionally, both ways are 100% equivalent.
>
> You can see for yourself that the implementation for rnws bundle uses the
> first variation:
> https://github.com/mjohnston/react-native-webpack-server/blob/168e0809cf14ade8aaf756bb95ec484fa386c2f8/bin/rnws.js#L146-L147.
> As you can see, this makes the HTTP request as soon as the server.start()
> promise resolves. My change is to make the server.start() promise resolve
> when both the RN packager *and* the webpack server are ready to accept
> requests. Because if we open the request sooner and the RN packager takes a
> really long time to start up, we might risk timing out the incoming request
> because we can't respond to it soon enough.
>
> but it seems the only functional change is the timeout, which I'd argue is
> the responsibility of the CI system.
>
> How do you propose to do that? In what way can the CI system take
> responsibility here?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mjohnston/react-native-webpack-server/pull/144/files#r46076347>
> .
>